### PR TITLE
Prompt user for signing scenarios rule scope

### PR DESCRIPTION
**Pull request by jgeurten**
1. Edited the WinSiPolicy 
2. Default UI if policy supports UMCI: 

image

3. Default UI if policy does not have UMCI on: 

image

4. Selecting the UMCI rule scope checkbox will ask the user whether they want the Wizard to set UMCI: 

image

### CODE REVIEW

**Comment by jsuther1974** [comment]
If the policy doesn't enable UMCI, couldn't we just change the rule default to kernel and gray out the user mode option rather than the warning message?

**Comment by jgeurten** [comment]
My thinking is that if someone is clicking the UMCI rule scope checkbox, they want the rule to apply to UMCI, so the Wizard should set UMCI in the policy

**Comment by jsuther1974** [comment]
Gotcha
________________________________

File: WDAC-Policy-Wizard/app/src/Helper.cs [hunk 1]
```diff
@@ -1467,38 +1485,86 @@ private static Signer[] SetSignersEKUs(Signer[] signers, EKU eku)
...
-                    refCopy.FileRuleRef[0].RuleID = ruleID;
-                }
-                else
+
+                for (int i = 0; i < siPolicy.SigningScenarios.Length; i++)
                 {
-                    refCopy.FileRuleRef = new FileRuleRef[siPolicy.SigningScenarios[1].ProductSigners.FileRulesRef.FileRuleRef.Length + 1];
-                    for (int i = 0; i < refCopy.FileRuleRef.Length - 1; i++)
+                    // Kernel mode (131)
+                    if (siPolicy.SigningScenarios[i].Value == 131 && scenarioStates.kmciEnabled)
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/src/Helper.cs:1506]: define the constant somewhere

File: WDAC-Policy-Wizard/app/src/CustomRuleConditionsPanel.Designer.cs [hunk 2]
```diff
@@ -30,6 +30,10 @@ private void InitializeComponent()
         {
```
[the remark this answers is not in the document]
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/src/CustomRuleConditionsPanel.Designer.cs:30]: Exactly. All the .Designer.cs are automatically generated

File: WDAC-Policy-Wizard/app/src/Helper.cs [hunk 3]
```diff
@@ -926,7 +926,7 @@ public static SiPolicy CreateAllowHashRule(SiPolicy siPolicy, PolicyCustomRules
                 cFileAllowRules++;
```
[the remark this answers is not in the document]
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/src/Helper.cs:926]: We need to sit down and go over the architecture and some possible redesigns

File: WDAC-Policy-Wizard/app/Properties/Resources.resx [hunk 4]
```diff
@@ -465,4 +465,10 @@ Please select the  'Create Rule' button.</value>
   <data name="WindowsTemplate" xml:space="preserve">
     <value>DefaultWindows_Audit.xml</value>
   </data>
+  <data name="InvalidKMCIRule" xml:space="preserve">
+    <value>Kernel mode components cannot be allowed or denied using packaged app or path rules.</value>
+  </data>
+  <data name="InvalidSigningScenarioCheckboxState" xml:space="preserve">
+    <value>At least one of Usermode Rule or Kernel Rule checkboxes must be checked</value>
+  </data>
 </root>
```
[the remark this answers is not in the document]
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/Properties/Resources.resx:474]: pshh

**Pull request merged**